CMakeLists.txt

```diff
@@ -28,6 +28,7 @@ add_library(NESO-Particles INTERFACE)
 # Alias the name to the namespaces name. Can use in subdirectory or via Confiig
 # files with namespace
 add_library(NESO-Particles::NESO-Particles ALIAS NESO-Particles)
+target_compile_options(NESO-Particles INTERFACE "-Werror;-Wpedantic;-Wall;-Wextra")
```

Comment:
I don't think I want this, particularly `-Werror`, because what happens is a user uses a compiler we haven't tested with and gains a new warning and now it fails to build which creates unnecessary fustration/friction to actually getting things to run.

I you want to have these enabled by say but a develop flag then sure.